server/src/main/java/org/elasticsearch/index/mapper/MetadataFieldMapper.java

```diff
@@ -129,6 +134,42 @@ public final MetadataFieldMapper build(MapperBuilderContext context) {
             return build();
         }
 
+        private static final Set<String> UNSUPPORTED_PARAMETERS_8_6_0 = Set.of("type", "fields", "copy_to", "boost");
+
+        public final void parseMetadataField(String name, MappingParserContext parserContext, Map<String, Object> fieldNode) {
```

Comment:
This gets the job done for now. We may want to revisit FieldMapper.Builder, Mapper.Builder etc. and see whether we can make this nicer.